### PR TITLE
Implement mechanism for helpers to create new tabs for apps.

### PR DESCRIPTION
**Pull request by eteq**
This PR adds a methods to the `jdaviz` helpers that allow them to be popped out into new tabs in jupyterlab. This video shows the two new ways of doing it in action.

https://user-images.githubusercontent.com/346587/139784545-a1e9011c-b9f6-4263-a2eb-feb256e216c7.mp4

This works as-is, but I have several standing questions which might need implementation of something before I'm ready to call this done:

- [x] Right now the default for `show_in_sidecar` is to appear on the right. The `anchor` keyword lets you place the new tab basically anywhere - new tab before, new tab after (which is how `show_in_new_tab` works), or bottom/top split view.  Should I pick a different default?  Maybe even have it be a config-specific choice what the default is?  For example, specviz is probably better at the bottom because it's "wide", while cubeviz is more "square"... But that's a bit more work, and could always be added later.
- [x] Should we even have `show_in_new_tab` given it could also just be ``self.show_in_sidecar(anchor='tab-after')``?
- [x] Should I update *all* the example notebooks with the `show_*` options instead of just specviz?
- [x] Should I just make `sidecar` an dependency instead of the current "it doesn't work unless you have sidecar installed" error message?
- [x] Should this be implemented on the `Application` class instead of the base helper?  I feel strongly this needs to be in the helpers, *but* it might belong in the application and have the helper methods just be something like:
```
def show_in_sidecar(self, **kwargs):
    return self.app.show_in_sidecar(**kwargs)
```
- [x] Are these trivial unit tests sufficient testing for now?

### CODE REVIEW

**Review by kecnry — COMMENTED** [review]
Works beautifully!  I would love to see if we could get similar behavior outside of lab, but that can come down the road and I think this will already be hugely useful for a lot of cases.

> Right now the default for show_in_sidecar is to appear on the right. The anchor keyword lets you place the new tab basically anywhere - new tab before, new tab after (which is how show_in_new_tab works), or bottom/top split view. Should I pick a different default? Maybe even have it be a config-specific choice what the default is? For example, specviz is probably better at the bottom because it's "wide", while cubeviz is more "square"... But that's a bit more work, and could always be added later.

I think config-dependent defaults might be a good idea, but agree that can be added later based on feedback/etc.  Ideally those would belong in the yaml files with the rest of the layout defaults.

 > Should we even have show_in_new_tab given it could also just be self.show_in_sidecar(anchor='tab-after')?
 
 Its definitely easier to remember, but I have no strong opinion.
 
 > Should I update all the example notebooks with the show_* options instead of just specviz?
 
 My vote would be yes, as it then suggests there are other alternatives and also feels less like "notebook magic".
 
 > Should I just make sidecar an dependency instead of the current "it doesn't work unless you have sidecar installed" error message?
 
I think the current import error statement is sufficient and we can always reconsider if it becomes the default usage or get feedback that its confusing.

> Should this be implemented on the Application class instead of the base helper? I feel strongly this needs to be in the helpers, but it might belong in the application and have the helper methods just be something like:

No opinion here as long as the code only lives in one place and don't need to call `viz.app.show_*`

**Comment by pllim** [comment]
No time to take this for a spin but I like the concept. Thanks!

You do need to address the failing CI. Some comments/questions below.

1. What happens when someone runs these in a non-Lab setup? Do they get exception, warning, or it is a silent no-op? 
2. If you really want this to be in helper, can it at least be in a higher level parent class, so all helpers can inherit?
3. `anchor` keyword is a good idea but we should keep the options simple. I will never remember `'tab-after'` but I will remember top/bottom/left/right.

**Comment by duytnguyendtn** [comment]
Interesting behavior I saw when I combined `show_in_new_tab()` and `show_in_sidecar()`. The two views are duplicated (as expected since they are **the same app** but when I open the Data drop down, I get three menus??

image

FWIW I do NOT have it open in the original notebook tab, so I only explicitly launched two specviz views

**Review by duytnguyendtn — COMMENTED** [review]
Before I get to answers, see #965 as a potential solution to ~all~ many of our problems here :)

* I think `anchor` would be useful EVENTUALLY, but I don't think it should block this PR. I think this adds such a nice quality of life improvement that it shouldn't wait on that
* See #965 as the solution here! With a `show()` method, we could add different arguments to control how it's viewed! (no need for separate functions)
* See #965 
* Yes, I would say `sidecar` should be a dependency with this PR
* I defer to your strong opinions here :)

As for blockers, I agree with Pey Lian's review: https://github.com/spacetelescope/jdaviz/pull/952#issuecomment-959143842

**Comment by eteq** [comment]
Alright, based on the above I plan to implement the following:
* Won't worry about the default right now, because we can update it later based on feedback and/or #965 
* Keep `show_in_new_tab ` but #965 might redner it unnecessary
* Will update all the example notebooks
* Will add sidecar as a dep
* Will fix the CI.

Now for @pllim's items:

> What happens when someone runs these in a non-Lab setup? Do they get exception, warning, or it is a silent no-op?

Yeah, good question: it's a pretty sensible fallback here it turns out (inherited from `Sidecar`): in notebook either of these shows the same thing `.app` shows - in the output cell you get the UI. So in user notebooks it become safe to just do `specviz.show_in_new_tab()` and in the notebook it will still show the UI, just not in a new tab. (P.S. This is yet another argument for adding this to #965, though, because then `.show()` just always yields an output in the notebook that's the UI even if you asked it for a new tab or whatever.)

 
> If you really want this to be in helper, can it at least be in a higher level parent class, so all helpers can inherit?

It's now in the base helper, so it should be in all of the helpers.  I'll expand the example notebooks to test that though to be sure.

> anchor keyword is a good idea but we should keep the options simple. I will never remember 'tab-after' but I will remember top/bottom/left/right.

Good points, but I agree with @duytnguyendtn this is better dealt with in `show` since that would be easier to just clearly say "right"/"left" etc as you suggest.

**Comment by eteq** [comment]
Alright, I think this has now addressed all the comments (with the caveat that some of them are "fix in #965 instead").

W.r.t. #965, I do agree with @duytnguyendtn it cleans up the interface a fair bit.  I propose we merge this ASAP and then have that change this machinery to use the `show`-oriented API. If we agree with that plan I'll leave a comment in #965 what I think the steps are that would be needed there to turn this into `show`.

File: CHANGES.rst
Comment (rosteen) [inline CHANGES.rst:7]:
```suggestion
- There are now `show_in_sidecar ` and `show_in_new_tab` methods on all the
```

File: notebooks/CubevizExample.ipynb
Comment (rosteen) [inline notebooks/CubevizExample.ipynb:66]:
```suggestion
    "viz.show_in_new_tab()"
```

**Review by rosteen — COMMENTED** [review]
Left suggestions for a couple spelling fixes. I like this, but I noticed that in both the sidecar and new tab cases, if the browser window isn't tall enough to show the whole vertical height of the app, it just gets cut off. I would have expected a scroll bar to be able to see the rest of the app, but I also suspect this is a sidecar/lab issue and not something we can control.

File: jdaviz/tests/test_show.py
Comment (rosteen) [inline jdaviz/tests/test_show.py:1]:
```suggestion
from jdaviz import Specviz
```

**Comment by eteq** [comment]
Thanks @rosteen - I'll take a moment later if I can find any quick-fix for the scrolling, but I agree it's probably on jupyterlab itself and not really something we can fix here.

**Comment by eteq** [comment]
Ok I did a deeper dive than I expected, and I am tantilizingly close to a solution to the scrolling problem. To summarize what I've learned:

* The fix would be to add the `overflow: auto` css attribute to the relevant `div`.  *however* it is highly non-trivial *which* div needs this attribute - doing some random setting of layout attributes inside the notebook led to the wrong behavior
* There is a div with class `lm-Widget p-Widget lm-Panel p-Panel jp-OutputArea-child` that seems to be in all the displays, and if I manually set the css attribute `"overflow"` to `"auto"`, we get exactly the scroll bar behavior we want (hidden until the view are is not wide enough).  I don't know how general this is, but it didn't show the vertical scrollbar in specviz at least in the notebook, but in the sidecar/tab, it did appear if I shrunk the window.
* However, I have no idea where that particular div is defined.  It isn't anywhere I could obviously reach from Python, although it could be buried in vue somewhere.  But this *might* require some changes in one of the vue templates or something?
* Regardless, this is the exact problem we see with *horizontal* scrolling in the notebook - if your jupyterlab window is too narrow there's no way to scroll it, which has long been a mild irritation for me when on a small screen.  This means I think the fix for this is *not* part of this PR, since it seems to be in the ipyvuetify machinery rather than the sidecar.

If that sounds good, @rosteen, I think this is ready to go, and the above can go in a separate issue.  If anyone (@mariobuikhuizen seems most likely?) has insight where that specific div gets set that would be awesome, although we can discuss that in a follow-on issue.

File: jdaviz/core/helpers.py
Comment (pllim) [inline jdaviz/core/helpers.py:299]: Move this before "Parameters". It is not rendering properly in https://jdaviz--952.org.readthedocs.build/en/952/api/jdaviz.core.helpers.ConfigHelper.html#jdaviz.core.helpers.ConfigHelper.show_in_sidecar

Also maybe give a direct link to the said documentation.

File: jdaviz/core/helpers.py
Comment (pllim) [inline jdaviz/core/helpers.py:313]: Unnecessary. You already made it a required dependency in `setup.cfg`.

```suggestion
        Notes
        -----
        If this method is called in the "classic" Jupyter notebook, the app will
        appear inline, as only lab has a mechanism to have multiple tabs.
```

File: jdaviz/core/helpers.py
Comment (pllim) [inline jdaviz/core/helpers.py:325]:
```suggestion
        from sidecar import Sidecar
```

File: notebooks/CubevizExample.ipynb
Comment (pllim) [inline notebooks/CubevizExample.ipynb:48]:
```suggestion
    "The above shows the Cubeviz UI inline with the notebook.  If you are in JupyterLab you can instead have the Cubeviz UI appear in a separate tab or as a \"sidecar\" (a split UI pane to the side of the notebook).  The below should do this (but only in JupyterLab.  If you are in the \"classic\" notebook it will show the same thing as `viz.app`)"
```

File: notebooks/MosvizExample.ipynb
Comment (pllim) [inline notebooks/MosvizExample.ipynb:271]:
```suggestion
    "The above shows the Mosviz UI inline with the notebook.  If you are in JupyterLab you can instead have the Mosviz UI appear in a separate tab or as a \"sidecar\" (a split UI pane to the side of the notebook).  The below should do this (but only in JupyterLab.  If you are in the \"classic\" notebook it will show the same thing as `mosviz.app`)"
```

File: notebooks/SpecvizExample.ipynb
Comment (pllim) [inline notebooks/SpecvizExample.ipynb:60]:
```suggestion
    "Note: you probably want to pick one of the three options below depending on preference. Also not the latter two will only work in JupyterLab, not the \"classic\" notebook interface."
```

File: jdaviz/core/helpers.py
Comment (pllim) [inline jdaviz/core/helpers.py:338]:
```suggestion
        Display the Jdaviz application in a new tab in JupyterLab.
```

File: notebooks/ImvizExample.ipynb
Comment (pllim) [inline notebooks/ImvizExample.ipynb:197]:
```suggestion
    "The above will show Imviz inline in the notebook.  The following two cells are alternative ways to show it in separate windows, either on the right side of the notebook or in a new tab.  (Note this will only work in JupyterLab, *not* the \"classic\" notebook.)"
```

File: notebooks/SpecvizExample.ipynb
Comment (pllim) [inline notebooks/SpecvizExample.ipynb:99]:
```suggestion
    "This will open specviz in a new JupyterLab tab:"
```

File: jdaviz/core/helpers.py
Comment (pllim) [inline jdaviz/core/helpers.py:344]:
```suggestion
            application; e.g., "specviz".
```

File: jdaviz/core/helpers.py
Comment (pllim) [inline jdaviz/core/helpers.py:347]: Move this before Parameters.
```suggestion
        Additional keywords are passed into the ``sidecar.Sidecar`` constructor.
        See ``jupyterlab-sidecar`` for the most up-to-date options.
```

Also maybe give a direct link to the said documentation.

File: jdaviz/core/helpers.py
Comment (pllim) [inline jdaviz/core/helpers.py:358]:
```suggestion
```

File: jdaviz/core/helpers.py
Comment (pllim) [inline jdaviz/core/helpers.py:369]: Maybe it is nicer to fallback to default?
```suggestion
                warnings.warn('show_in_new_tab cannot have a non-tab anchor, using tab-after', UserWarning)
                kwargs['anchor'] = 'tab-after'
```

File: jdaviz/tests/test_show.py
Comment (pllim) [inline jdaviz/tests/test_show.py:1]:
```suggestion
```

File: jdaviz/tests/test_show.py
Comment (pllim) [inline jdaviz/tests/test_show.py:5]: Use pre-defined fixtures in `conftest.py`.
```suggestion
def test_show_inline(specviz_app):
```

File: jdaviz/tests/test_show.py
Comment (pllim) [inline jdaviz/tests/test_show.py:7]:
```suggestion
    res = specviz_app.show_inline()
```

File: jdaviz/tests/test_show.py
Comment (pllim) [inline jdaviz/tests/test_show.py:12]:
```suggestion
def test_show_sidecar(specviz_app):
```

File: jdaviz/tests/test_show.py
Comment (pllim) [inline jdaviz/tests/test_show.py:14]:
```suggestion
    res = specviz_app.show_in_sidecar()
```

File: jdaviz/tests/test_show.py
Comment (pllim) [inline jdaviz/tests/test_show.py:18]:
```suggestion
def test_show_new_tab(specviz_app):
```

File: jdaviz/tests/test_show.py
Comment (pllim) [inline jdaviz/tests/test_show.py:20]:
```suggestion
    res = specviz_app.show_in_new_tab()
```

File: notebooks/SpecvizExample.ipynb
Comment (pllim) [inline notebooks/SpecvizExample.ipynb:67]:
```suggestion
    "This option will show Specviz inline in the notebook"
```

**Review by pllim — COMMENTED** [review]
@havok2063 , do you care about this feature? Does this break your config injection workflow?

I don't know how I feel about putting all these in the notebook and then say "well, don't do all these at once". I feel like the examples belong more in "concepts". However, in the official example, why not use `...viz.show_inline()` instead of `...viz.app`?

While possible, I don't see any benefit telling Cubeviz users about the sidecar feature. Everything is just smooshed. (Debatable for Mosviz and Specviz, though I can see it being very useful for a simple case of Imviz.)

**Comment by pllim** [comment]
> I think this is ready to go

Are you sure? tl;dr but seems like there is something non-trivial that still needs fixing. Should we at least document that under Known Issues?

**Comment by pllim** [comment]
Re: tests -- I agree it is hard to test the GUI part. I think these are sufficient for now until we figure out GUI testing.

**Comment by pllim** [comment]
p.s. A missed opportunity you didn't name this branch `pop-life` https://www.youtube.com/watch?v=fxsTGequ1do

**Comment by havok2063** [comment]
Coming late to this thread, but on the MAST side, I'm not sure we'll need or use this feature.   It seems to me the point is to break out Jdaviz to provide it more screen space, while still being able to programmatically do stuff.  Since in MAST, users can already expand Jdaviz to the full screen, and there is no avenue for programmatic commands, I don't think this is currently that useful for MAST users.  

That being said, I could see a cool feature, if users could pop out individual viewers into a new browser tab or window, and have them all be linked.  I know `goldenlayouts` has this feature but I'm not sure if that is currently feasible in Jdaviz, or if this enables that?  

I am strongly in favor though of making this an opt-in dependency (i.e. `pip install jdaviz[.extras]`).  I don't think we want this installed by default, and certainly if MAST isn't using it we don't want to install unnecessary packages in our environments.  I also worry about modifying the css or config YAML files.  I'd want to make sure any css changes don't affect MAST, as well as any config settings are items MAST can easily turn off, which requires updates on our side. 

I do like the idea of a `show` method, but this feature would require some updates to our MAST hooks, so it'd great to coordinate on when this feature gets included.

File: CHANGES.rst
Comment (eteq) [reply CHANGES.rst:7]: Too many mardown comments lately 🐑

File: jdaviz/core/helpers.py
Comment (eteq) [reply jdaviz/core/helpers.py:296]: Good point, will fix in a follow-on commit

File: jdaviz/tests/test_show.py
Comment (eteq) [reply jdaviz/tests/test_show.py:5]: Thanks, did not know about the fixtures!

**Comment by eteq** [comment]
> Are you sure? tl;dr but seems like there is something non-trivial that still needs fixing. Should we at least document that under Known Issues?

It's been a  long-standing limitation in `jdaviz` already.  Basically if your screen isn't wide enough there's no solution.  I've been battling this on one particular computer that has a weird shape screen.  It's exacerbated in the sidecar but *only if you have a relatively narrow screen*.  My at-desk monitor, for example, which is widescreen, has no problem fitting it at default resolution.

Basically I'm saying this is not a problem *this* PR needs fixing.  It should be fixed, but the solution is independent of the implementation in this PR

**Comment by eteq** [comment]
> I am strongly in favor though of making this an opt-in dependency 

@havok2063 - can you clarify the downside of this?  I don't really care that strongly so I can back it out, but @pllim wanted it to be a dependency (and I don't really mind because sidecar is pretty easy to install, at least in my experience).  

I agree it shouldn't leave traces in the CSS, but I would have thought this wouldn't affect the mast layouts as long as you avoid using either of new tab/sidcar in the viz tools in MAST...?

**Comment by eteq** [comment]
(also I addressed *most* of @pllim's comments, but need to do a few more locally.  Will try to do that tomorrow.)

**Comment by kecnry** [comment]
regarding the dependency: would a compromise be to list it as a pip-dependency so users get it out-of-the-box without additional steps, but try/except the import so that MAST or advanced users can build from source and run jdaviz without it?

**Comment by havok2063** [comment]
@eteq I'm always in favor of keeping a package lean with its dependencies.  I'm just thinking of the case if `sidecar` is a "heavy" dependency, i.e something with a lot of other package baggage.  In which case we wouldn't want it as a core dependency automatically installed.  I also think of this feature as not a core feature asked for by the user, but as an extra nice-to-have fancy feature.  But I don't have a strong enough opinion on it to care that much.  

Yeah I don't think the CSS should affect MAST.  I'm just trying to think about all potential gotchas to cover all the bases.

**Comment by mariobuikhuizen** [comment]
@eteq I fix the scrolling issue with:
```css
.jupyterlab-sidecar .jp-OutputArea-output {
  overflow: auto;
}
```
which can be added to the `<style>` section of `app.vue`

This is indeed not reachable from python.

To make the scrolling also work in lab, the following can be used instead:
```css
.jp-OutputArea-child .jp-OutputArea-output {
  overflow: auto;
}
```

Note: setting `overflow` on `.jp-OutputArea-output` should have been enough (and in fact it already has this rule), but it's overridden by `.jupyter-widgets`, by adding a class to the selector (`.jp-OutputArea-child`) our selector is now more specific and overrides that again.

**Comment by pllim** [comment]
Re: https://github.com/spacetelescope/jdaviz/pull/952#issuecomment-1021182873 -- Might be a bit too late for this package to have any "lean dependencies" requirements. Have you seen this?

https://github.com/spacetelescope/jdaviz/blob/32d42431c22274702178be71a42d81ea06bc8e6c/setup.cfg#L20

Also see #399

File: jdaviz/core/helpers.py
Comment (eteq) [reply jdaviz/core/helpers.py:299]: original motivation was that sidecar does not have a proper docs page to link to... but now that you point this out I can point to the github repo which is the closest thing it has - will fix.

**Comment by eteq** [comment]
Thanks @mariobuikhuizen! Adding that to `app.vue` resolved the scrolling issue perfectly! I added it in the commit I just pushed up on the theory that we don't know exactly when the next `ipywidgets` will come out. If/when this PR is merged we can make another issue to revert that in favor of a bumping of requirements once it's out.

I think I've addressed all of inline  @pllim's comments except for one (which is 50% "disagree" and 50% "it won't matter anyway if we switch to `show()` as I think we all want to after this PR is merged.)

Re: @pllim's
> While possible, I don't see any benefit telling Cubeviz users about the sidecar feature. Everything is just smooshed. (Debatable for Mosviz and Specviz, though I can see it being very useful for a simple case of Imviz.)

I disagree there: on my current primary work monitor, if I expand my jupyterlab to cover the whole screen and evenly split the notebook and sidecar in jupyterlab, cubeviz is just the right size.  I think the reality here is that it depends greatly on your specific monitor/resolution combination, and the point is to show the users these options and leave it to them to decide which works best for them.


If we're ok with all that, the one remaining question is then the dependency issue.  @havok2063, I definitely take your point about "heavy" dependencies.  But my read of this is that `sidecar` is definitely not that - it's a very simple thing (https://github.com/jupyter-widgets/jupyterlab-sidecar/blob/master/sidecar/sidecar.py and https://github.com/jupyter-widgets/jupyterlab-sidecar/blob/master/src/widget.ts are the implementation, and that's only ~70 lines of code including both Python and javascript/typescript).  So I'm in favor of leaving this as it is based on @pllim's argument coupled with the fact that a *lot* of hack day users have actually requested something like this.

If we're OK with that, then I think this is ready to merge.

**Comment by pllim** [comment]
I'll approve when CI is green, unless @havok2063 has objection w.r.t. new dependency.

**Comment by eteq** [comment]
(oh, and for full disclosure,  7e150c9a95b48024e3bd77d0355285af35ad2336 is a change to ``.gitignore`` that makes my life easier w.r.t. virtual environment management. It's simple enough it seems silly to do as a stand-alone PR but isn't strictly related to this 🤷 )

**Comment by havok2063** [comment]
@eteq This seems reasonable to me.  It seems light enough then to include as is.  And if hack-day users are requesting it, then I think it should be an included feature.  So I nominally approve this.  If it's not too much trouble, and if you have the time, could you do a test with https://github.com/havok2063/standalone_jdaviz to see that it doesn't break anything with the default embed implementation?  I think it should be fine but it can't hurt to check.

**Review by pllim — APPROVED** [review]
LGTM but we should wait to see how https://github.com/spacetelescope/jdaviz/pull/952#issuecomment-1022438540 turns out first before merge.

**Comment by eteq** [comment]
Alright, so I followed the directions in https://github.com/havok2063/standalone_jdaviz  but with voila running in the same environment as an install of this branch. The example page on `standalone_jdaviz` worked just fine without any obvious issues.  Is that the test you had in mind @havok2063, or was there a specific thing you wanted me to check is still working?

**Comment by havok2063** [comment]
@eteq yes I think so.  As long as it ran without incident, and functionality is still intact.  Are there any changes to the Jdaviz UI with this PR that show up in the web, e.g. new buttons, etc that we need to worry about?  If it's only programmatically accessible, then it sounds like everything is a-ok.  I'm ok with merging this, and we can double check things on our end.

**Comment by eteq** [comment]
That's right, @havok2063, no new buttons or other interface elements.  So I think we're all good there.

One thing that I thought might be affected but in fact was *not* is the change here: https://github.com/spacetelescope/jdaviz/pull/952/commits/7e572a1aea90a96d51cb95a155922062b04b065d
in retrospect it makes sense because it's got `jupyterlab` in the name , but point is: no unintended consequences there.

So I think this is good to go *except* for that one hanging test that ran for 6h and then timed out.  Does anyone know what's going on there?  I don't really see what it could have to do with this PR, but I've said that before and been very wrong...

**Comment by pllim** [comment]
> that one hanging test that ran for 6h and then timed out

I restarted the CI. If it still times out, I'll restart one at main branch and see. We should make sure before merge. Thanks for pointing that out!

**Comment by pllim** [comment]
Ugh, actually I think I see the remote data job taking very long in other PRs too. I can't tell if this is transient or something needs fixing. I'll give it a few days.

**Comment by eteq** [comment]
Great work applying the "maybe if I wait it'll fix itself" solution, @pllim, it seems to have done the trick! 😉

**Comment by pllim** [comment]
What, how?! I still see it in other jobs. 🤯

**Review by kecnry — APPROVED** [review]
Just to clarify - is the plan to merge this and then rebase #965 (@duytnguyendtn) to then "consolidate" all the show methods at that time?  And possibly even work #977 into that same framework?  Or does it make sense to hold off on merging and handle these all in a single PR instead?

**Comment by eteq** [comment]
@kecnry - @duytnguyendtn is welcome to have an alternative plan, but my proposal is to 
1. merge this ASAP to lower the number of intertwined PRs while making sure we are making progress on the actual feature
2. rebase #965 over this, and from there consider some of the interface questions like a single `show` method instead of multiple methods, etc.
3. work on #977 like it is right now in parallel, *without* any helper method, instead being exposed only in the UI/javascript
4. Have another PR add #977 to the `show` methods (or possibly rebase #965 over that depending on how it's going)

**Comment by eteq** [comment]
FYI, this was previously all-green, not sure why suddenly that one problematic test got re-run (which continues to hang, it looks like).

**Comment by pllim** [comment]
Please rebase to pick up #1052 . Thanks!

**Comment by eteq** [comment]
rebased!

**Comment by pllim** [comment]
Thanks! @duytnguyendtn will do one last look and then merge.

**Comment by duytnguyendtn** [comment]
I agree with the plan described in https://github.com/spacetelescope/jdaviz/pull/952#issuecomment-1024506439! Looks like we have two approvals, so I'll formally merge this in and I'll also go back and rebase #965

**Pull request merged**
